src/Dependencies/Updater.php

```diff
@@ -12,13 +12,62 @@ public function __construct(
         $this->dir = $dir;
     }
 
-    public function update()
+    public function update($dep = null)
     {
         $result = $this->loadWhippetFiles();
+
+        if ($result->isErr()) {
```

Comment:
I think this is an an antipattern, but I haven't found a good link. It's kind of like '[Flag argument](https://www.martinfowler.com/bliki/FlagArgument.html)', but not quite the same.

What's happening here is that based on whether an argument is provided or not this function behaves quite differently. 

That's a sign that there may be actually two methods trying to get out - possibly one defined in terms of the other: one which accepts an argument, and one with no arguments. 

Consider making this distinction at the level where we're making sense of the arguments we've been passed - *(e.g. in `src/Modules/Dependencies.php`), and calling one or other method from there.